cmd/serve.go

```diff
@@ -100,13 +98,14 @@ func elementsHandler(config httpConfig, w http.ResponseWriter, r *http.Request)
 		return &httpError{err, "Error collecting elements", 500}
 	}
 
-	formattedOutput, outputErr := output.Generate(collectedElements)
-
 	title := fmt.Sprintf("Elements %s", version)
 	w.Header().Set("Server", title)
+
+	formattedOutput, outputErr := output.Generate(collectedElements)
+
 	if outputErr != nil {
 		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Invalid format requested"))
+		w.Write([]byte(outputErr.Error()))
```

Comment:
Seems sensible to me to let the error itself do the talking here, since it may change or grow to encompass other types of error conditions.